typescript/cli/src/tests/warp-apply-zksync.e2e-test.ts

```diff
@@ -0,0 +1,337 @@
+import { expect } from 'chai';
+import { Wallet } from 'ethers';
+
+import { ChainAddresses } from '@hyperlane-xyz/registry';
+import {
+  TokenRouterConfig,
+  TokenType,
+  WarpRouteDeployConfig,
+} from '@hyperlane-xyz/sdk';
+
+import { readYamlOrJson, writeYamlOrJson } from '../utils/files.js';
+
+import {
+  ZKSYNC_KEY,
+  deployOrUseExistingCore,
+  extendWarpConfig,
+  getChainId,
+  updateOwner,
+} from './commands/helpers.js';
+import {
+  hyperlaneWarpApply,
+  hyperlaneWarpDeploy,
+  readWarpConfig,
+} from './commands/warp.js';
+
+const CHAIN_NAME_ZK_2 = 'zksync1';
+const CHAIN_NAME_ZK_3 = 'zksync2';
+
+export const TEST_CONFIGS_PATH = './test-configs';
+export const ZK_REGISTRY_PATH = `${TEST_CONFIGS_PATH}/zksync`;
+
+const BURN_ADDRESS = '0x0000000000000000000000000000000000000001';
+const EXAMPLES_PATH = './examples';
+const CORE_CONFIG_PATH = `${EXAMPLES_PATH}/core-config-zksync.yaml`;
+const WARP_CONFIG_PATH_EXAMPLE = `${EXAMPLES_PATH}/warp-route-deployment-zksync.yaml`;
+
+const TEMP_PATH = '/tmp'; //temp gets removed at the end of all-test.sh
+const WARP_CONFIG_PATH_2 = `${TEMP_PATH}/zksync/warp-route-deployment-zksync.yaml`;
+const WARP_CORE_CONFIG_PATH_2 = `${ZK_REGISTRY_PATH}/deployments/warp_routes/ETH/${CHAIN_NAME_ZK_2}-config.yaml`;
+
+const TEST_TIMEOUT = 180_000; // Long timeout since these tests can take a while
+describe('WarpApply zkSync e2e tests', async function () {
```

Comment:
think you'll have to skip the warp apply + warp read e2e tests for now, because:
1. the test zksync chain metadata does not match that of the anvil networks we spin up
2. anvil networks don't really test zksync functionality anyway

```suggestion
describe.skip('WarpApply zkSync e2e tests', async function () {
```

so i'd say we keep the tests in, but skip here + in `warp-read-zksync.e2e-test.ts`, until we have the zksync e2e tests use https://docs.zksync.io/build/test-and-debug/in-memory-node

Reply:
For testing zksync e2e I had to use both in-memory node and a dockerized version of zksync, since the in-memory node doesn't allow for custom chainId to be set [default is chainId: 260 on all in-memory node instances], if the same node process was used it would lead to chainId/domainId clash error

Reply:
Yeah, that makes complete sense